src/main/java/com/wavefront/fdb/utils/BatchReader.java

```diff
@@ -69,26 +80,37 @@
    * Interface to call in order to generate metrics for batch reader operations.
    */
   private final Metrics metrics;
+  /**
+   * Max wait time in backoff algorithm in milliseconds.
+   */
+  private final long maxWaitTime;
+  /**
+   * Single Thread Executor to schedule retry task.
+   */
+  private final ScheduledExecutorService ses = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Also, can we not declare our own SES (that's the part that I was trying to avoid with jdk 11)